pkg/security/security.go

```diff
@@ -351,12 +352,41 @@ type Caller struct {
 	Identities []string
 }
 
+// AuthResult is the result of authentication action.
+type AuthResult struct {
```

Comment:
confusing, after first look i thought it is the return value of `Authenticate()`